core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -102,6 +107,27 @@ case class Uri(
   override protected def self: Self = this
 
   override protected def replaceQuery(query: Query): Self = copy(query = query)
+
+  /**
+   * Converts a [[Uri]] into an [[InetSocketAddress]] as long as this [[Uri]] has an authority.
+   * This conversion is a pretty straightforward translation, most of the work is inserting default port values
+   * if the [[Uri]] does not specify explicit ports.
+   * This method only has knowledge of the Http protocol (including websockets), it does not know how to guess ports
+   * for schemes it has no knowledge of and will return an [[IOException]] if it encounters an unknown scheme.
+   * If the uri does not explicit a scheme, it will assume the [[Http]] scheme.
+   */
+  def asAddress: Throwable \/ InetSocketAddress = {
+    val assumedScheme = scheme.getOrElse(Http)
+    val defaultPort = assumedScheme match {
+      case Http | Ws => Some(80)
+      case Https | Wss => Some(443)
+      case _ => None
+    }
+    for {
+      auth <- authority \/> new IOException("Request must have an authority")
+      port <- port.orElse(defaultPort) \/> new IOException("Unknown scheme")
+    } yield new InetSocketAddress(auth.host.value, port)
```

Comment:
Well, this code is extracted from `Http1Support` where this constructor was used, presumably resulting in a reverse DNS lookup like you point out. I am not too sure what is desired here, my motivation was to make this function available outside of that class as well as add support for the websocket scheme but otherwise leave the functionality intact.